src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -91,443 +84,251 @@ internal class SqlServerFhirDataStore : IFhirDataStore, IProvideCapability
+
+            await MergeResourcesCommitTransactionAsync(minSurrId, cancellationToken);
+
+            return results;
         }
 
-        // TODO: Remove when Merge is min supported version
-        private void PopulateUpsertResourceCommand(
-            SqlCommandWrapper sqlCommandWrapper,
-            ResourceWrapper resource,
-            bool keepHistory,
-            int? comparedVersion,
-            RecyclableMemoryStream stream,
-            bool isResourceChangeCaptureEnabled)
+        internal async Task TryLogEvent(string process, string status, string text, DateTime? startDate, CancellationToken cancellationToken)
         {
-            long baseResourceSurrogateId = ResourceSurrogateIdHelper.LastUpdatedToResourceSurrogateId(resource.LastModified.UtcDateTime);
-            short resourceTypeId = _model.GetResourceTypeId(resource.ResourceTypeName);
-
-            // NOTE: in the following code a call to autogenerated method _upsertResourceTvpGeneratorV[Version].Generate() then appears to be making a
-            // number of calls to interface ITableValuedParameterRowGenerator.GenerateRows() methods. But GenerateRows() contains "yield return"
-            // statements, so GenerateRows() body is not really executed at this point and no rows are generated, only IEnumerable interface is returned.
-            // This comment is put in here instead of in _upsertResourceTvpGeneratorV[Version].Generate() because
-            // _upsertResourceTvpGeneratorV[Version].Generate() is autogenerated by a tool so we cannot put comment there.
-            VLatest.UpsertResource.PopulateCommand(
-                sqlCommandWrapper,
-                baseResourceSurrogateId: baseResourceSurrogateId,
-                resourceTypeId: resourceTypeId,
-                resourceId: resource.ResourceId,
-                eTag: null, // not used in stored procedure
-                allowCreate: true, // not used in stored procedure
-                isDeleted: resource.IsDeleted,
-                keepHistory: keepHistory,
-                requireETagOnUpdate: true, // not used in stored procedure
-                requestMethod: resource.Request.Method,
-                searchParamHash: resource.SearchParameterHash,
-                rawResource: stream,
-                tableValuedParameters: _upsertResourceTvpGeneratorVLatest.Generate(new List<ResourceWrapper> { resource }),
-                isResourceChangeCaptureEnabled: isResourceChangeCaptureEnabled,
-                comparedVersion: comparedVersion);
+            try
+            {
+                using var conn = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, false);
+                using var cmd = conn.CreateNonRetrySqlCommand();
+                VLatest.LogEvent.PopulateCommand(cmd, process, status, null, null, null, null, startDate, text, null, null);
+                await cmd.ExecuteNonQueryAsync(cancellationToken);
+            }
+            catch
```

Comment:
Doesn't it make sense to at least log an exception using _logger?

Reply:
This is useful only for database centric testing. No.